src/libraries/BotBuilder/Microsoft.Agents.BotBuilder/UserAuth/UserAuthorizationDispatcher.cs

```diff
@@ -0,0 +1,191 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Agents.Authentication;
+using Microsoft.Agents.BotBuilder.Errors;
+using Microsoft.Agents.Core.Errors;
+using Microsoft.Agents.Core.Models;
+using Microsoft.Agents.Storage;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.Loader;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Agents.BotBuilder.UserAuth
+{
+    /// <summary>
+    /// Loads and dispatches request to a named instance of IUserAuthorization.
+    /// </summary>
+    /// <remarks>
+    /// This utilizes type loading to support extensibility.
+    /// </remarks>
+    public class UserAuthorizationDispatcher : IUserAuthorizationDispatcher
```

Comment:
Public or internal?

Reply:
Depends.  I'm open.  Someone could use this outside AgentApplication.  It doesn't have any notion of app behavior.  It rather like IConnections.  Though I can't think of a use-case... In theory someone could use it from, say, ActivityHandler.

Reply:
Internal now